opentracing-noop/src/main/java/io/opentracing/NoopTracer.java

```diff
@@ -15,19 +15,25 @@
 
 import io.opentracing.propagation.Format;
 
-public final class NoopTracer implements Tracer {
+public interface NoopTracer extends Tracer {
+    static NoopTracer create() {
+        return NoopTracerImpl.INSTANCE;
+    }
```

Comment:
The intention to the `create()` method was to make clear that the instance cannot be compared against. 

To avoid the mix-and-match problem you need code to check `span instanceof NoopSpan` and not `NoopSpan.INSTANCE == span`. The create method hopefully pushes the user into accepting this.